tester/runner.go

```diff
@@ -259,7 +266,9 @@ func (r *Runner) RunTests(ctx context.Context, txn storage.Transaction) (ch chan
 				if !strings.HasPrefix(string(rule.Head.Name), TestPrefix) {
 					continue
 				}
-				tr, stop := r.runTest(ctx, txn, module, rule)
+				runCtx, cancel := context.WithTimeout(ctx, r.timeout)
+				defer cancel()
+				tr, stop := r.runTest(runCtx, txn, module, rule)
```

Comment:
We should call `cancel` as soon as the current test finishes to release the resources associated with the timeout. As is this will call cancel() after ALL tests have executed. I'd probably wrap this in an anonymous function.

Reply:
Good catch, I have wrapped the call into an anonymous function.